### PR TITLE
Add hostpath.exe

### PR DESCRIPTION
**Pull request by joeloff**
`hostpath.exe` is a .NET 4.5 executable that runs as part of Visual Studio setup on non-x86 machines to remove the an potential x86 dotnet entries from the `PATH`.

The `PATH` variable is read directly from the registry to avoid expanding existing environment variables in the path - (`Environment.GetEnvironmentVariable` always expands). For example, on a clean machine you'd typically see an entry like `%SystemRoot%\system32;%SystemRoot%;%SystemRoot%\System32\Wbem;%SYSTEMROOT%\System32\WindowsPowerShell\v1.0\;%SYSTEMROOT%\System32\OpenSSH\;` which becomes `C:\Windows\system32;C:\Windows;...` when expanded. We need to avoid writing the expanded value back.

The package is statically versioned at 6.0.2 since VS currently consumes the 6.0.1 redist package.

### CODE REVIEW

**Comment by joeloff** [comment]
@NikolaMilosavljevic I'll probably push to the internal branch to test the signing/publishing steps for Visual Studio.

@MichaelSimons Do we need to exclude anything here for source build? This is Windows+VS only and not something that would be consumed by other repos.

**Comment by MichaelSimons** [comment]
@joeloff - thanks for keeping source-build in-mind.  Source-build already defines a [custom default subset](https://github.com/dotnet/deployment-tools/blob/main/eng/Subsets.props#L42) so no changes should be necessary.

File: src/hostpath/src/Program.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                    // Check for paths with and without a trailing backslash
+                    string x86DotNetPath1 = Path.Combine(
+                        Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86),
+                        "dotnet");
+                    string x86DotNetPath2 = Path.Combine(
+                        Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86),
+                        @"dotnet\");
+
+                    List<string> paths = path.Split(';').ToList();
+                    paths.Remove(x86DotNetPath1);
```
[the remark this answers is not in the document]
Comment (joeloff) [reply src/hostpath/src/Program.cs:45]: yup, this should be RemoveAll instead.

**Comment by joeloff** [comment]
@marcpopMSFT since we plan on removing this from VS, you still want me to merge this so we have a copy?

**Comment by joeloff** [comment]
@marcpopMSFT this got dropped. Do you think we still want to save a copy. We addressed the underlying issue in VS and the host installers and then removed the EXE from VS and all the versions that did ship it are now out of support.

**Comment by marcpopMSFT** [comment]
> @marcpopMSFT this got dropped. Do you think we still want to save a copy? We addressed the underlying issue in VS and the host installers and then removed the EXE from VS and all the versions that did ship it are now out of support.

We still get reports of this in VS but I think there is zero chance we bring it back. If you want to save the source code, you could stick it in a different repo (our internal scripts repo maybe) but we'll likely never need it. You could just put it in your own fork of a repo so it's saved on GH.

**Comment by joeloff** [comment]
Sounds good, I'll stick the code somewhere else in case we need it. VS may not be a bad idea since it was specifically for VS that we had the code.

**Pull request closed without merging**